packages/react/src/components/route-link/route-link.tsx

```diff
@@ -32,8 +32,13 @@ const Link = styled(StyledLink)<{ $isMobile: boolean, $hasLabel: boolean }>`
 `;
 
 type Nav = typeof NavLink;
+type LinkType = typeof ReactRouterLink;
 
-export interface RouterLinkProps extends Nav {
+export interface NavLinkRouterLinkProps extends Nav {
+    displayName?: string;
+}
+
+export interface LinkRouterLinkProps extends LinkType {
```

Comment:
C'est compliqué pour rien ça, je sais pas pourquoi c'est là, je pense pas que ça soit utile si tu fais les changements que je te propose.

Reply:
j'ai pas fait ça pour le fun. c'est pas mal la seule façon que j'ai trouvé de régler le problème de typings avec ReactRouterLinkProps. C'est le problème initial dans cpanel 2 que j'ai maintenant ici aussi. voici ce que ça donne quand je fais tout tes commentaires et que j'enlève ces changements là : 
<img width="1437" alt="Capture d’écran, le 2022-10-12 à 12 04 34" src="https://user-images.githubusercontent.com/19400226/195392893-ed226ab0-9873-4751-986c-3ddb31c5484a.png">
en gros, on peut pu passer Link à RouteLink maintenant : 
`<RouteLink routerLink={Link} href="/test" iconName="mail" />`
mais on peut encore passer NavLink : 
`<RouteLink routerLink={NavLink} href="/story1" label="Route Link" />`

Reply:
okey je viens de voir ton autre comment. ça va marcher avec type of je crois, mais je comprends pas trop pourquoi on avait ça dans le code par contre : 
```
export interface RouterLinkProps extends Nav {
    displayName?: string;
}
```

Reply:
ça fonctionne avec type of